ddtrace/internal/utils/http.py

```diff
@@ -175,6 +175,12 @@ def w3c_get_dd_list_member(context):
             # we've already added sampling decision and user id
             and k not in [SAMPLING_DECISION_TRACE_TAG_KEY, USER_ID_KEY]
         ):
+            # Hack: convert byte string to unicode to use the regex below.
+            if isinstance(k, bytes):
+                k = k.decode("ascii", "ignore")
+            if isinstance(v, bytes):
+                v = v.decode("ascii", "ignore")
```

Comment:
Very ugly fix. The W3C regex below expects a unicode string however http headers can contain bytes. 

Alternatives:
- Avoid decoding the byte string and handling decoding errors. Instead create a new regex which expects a bytes string.
Con: Handling two different string types can be messy. Using one type is easier to maintain.

Reply:
Does this need to be handled in this PR?

Reply:
Without this fix this test fails: https://github.com/DataDog/dd-trace-py/blob/eecef546917b5cbe99a22e3ebe57e58ab07a3095/tests/tracer/test_propagation.py#L68. 

I can update this test to only use the `datadog` propagation style. Fix the bug. Then revert this test back to testing the default propagation mode. If the fix does not make it into 1.7 then it will require a release note.

Reply:
Moved to: https://github.com/DataDog/dd-trace-py/pull/4785